### PR TITLE
Javascript functionalities Added

### PR DESCRIPTION
**Pull request by chandak876**
1. Clicking the Tagged tab, should open the login popup
2. Log In button will redirect to the login page
3. Sign Up button will redirect to sign up page
4. Clicking on Follow button toggles it to Unfollow.

Target Design
Target Design

Designs Created
Design Layout
Design Layout
Design Layout

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -134,7 +136,49 @@ <h4 class="profile-tabbar-tab-text">TAGGED</h4>
                 <p>3060</p>
             </div>
         </div>
+    </section>
+
+<!--Login Popup box section-->
+    <section class="login-popup" id="login-popup-modal">
```
Comment (harsh-rk) [inline index.html]: login-popup-modal <> login-popup

File: index.html [hunk 2]
```diff
@@ -134,7 +136,49 @@ <h4 class="profile-tabbar-tab-text">TAGGED</h4>
...
+            <input
+                    type="text"
+                    placeholder="Password"
+                    class="login-input"
+            />
+
+            <a href="#" class="login-button">Log In</a>
+
+            <div class="login-seperator">
+                <p class="login-seperator-line-first"></p>
```
Comment (harsh-rk) [inline index.html]: Please check if we can use flex on the container and rename this to `login-seperator-line`

File: loginPage.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        <p class="login-forgot-password">Forgot password?</p>
+    </section>
+
+<!--Sign up section-->
+    <section class="signup">
+        <center><p class="signup-text">Don't have an account? <span class="signup-button">Sign up</span></p></center>
+    </section>
+
+<!--Download app section-->
+    <section class="getapp">
```
Comment (harsh-rk) [inline loginPage.html]: "getapp" <> "download-section"

File: loginPage.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        <center><p class="getapp-text">Get the app.</p></center>
+
+        <<div class="inline">
+        <a href="https://itunes.apple.com/app/instagram/id389801252?pt=428156&ct=igweb.loginPage.badge&mt=8&vt=lo">
+            <img class="getapp-tag-appstore" src="./Images/app-store-icon.png">
```
Comment (harsh-rk) [inline loginPage.html]: getapp-tag-appstore <> download-appstore-image

File: loginPageStyle.css [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    --very-light-blue: #B3DFFC;
+    --light-grey: #FAFAFA;
+    --dark-grey: #A1A1A1;
+    --mild-grey: #DBDBDB;
+    --mildish-grey: #8E8E8E
+}
+
+html{
+    font-size: 62.5%;
+    height: 100%;
```
Comment (harsh-rk) [inline loginPageStyle.css]: Please check if we need this

File: loginPageStyle.css [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+.login-facebook{
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+}
+
+.login-facebook-icon{
+    length: 1.5rem;
```
Comment (harsh-rk) [inline loginPageStyle.css]: Difference between length, height and width?

File: loginPageStyle.css [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+.login-seperator-text{
+    font-weight: bold;
+    font-size: 1.3rem;
+    color: var(--mildish-grey);
+}
+
+.login-facebook{
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
```
Comment (harsh-rk) [inline loginPageStyle.css]: Please check if we can remove this since we are already applying margin

File: loginPageStyle.css [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+/*Component based styling*/
+.login{
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    background: white;
+    border: 0.1rem solid var(--mild-grey);
+    padding: 2rem 3rem;
+    width: 29rem;
```
Comment (harsh-rk) [inline loginPageStyle.css]: Please define one variable and consume in all the places

File: loginPageStyle.css [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+    align-items: center;
+    justify-content: center;
+    min-height: 100%;
+}
+
+a{
+    text-decoration: none;
+}
+
+.inline{
```
Comment (harsh-rk) [inline loginPageStyle.css]: inline <> download-stores
Please give fixed width and then use `justify-content: space-between`

File: loginPageStyle.css [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+    display: flex;
+    flex-direction: column;
+}
+
+.getapp-text{
+    font-size: 1.43rem;
+    margin-bottom: 2rem;
+}
+
+.getapp-tag-appstore{
```
Comment (harsh-rk) [inline loginPageStyle.css]: getapp-tag-appstore <> download-stores-store

File: signUpPage.html [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+        <input
+                type="text"
+                placeholder="Password"
+                class="signup-input"
+        />
+
+        <a href="#" class="signup-button">Sign up</a>
+
+        <p class="signup-termsandcondn">By signing up, you agree to our <b>Terms , Data Policy and Cookies Policy </b>.</p>
```
Comment (harsh-rk) [inline signUpPage.html]: signup-termsandcondn <> signup-terms-and-conditions

File: signUpPageStyle.css [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+    margin-top: 0.8rem;
+}
+
+.signup-description{
+    margin-top: 0.91rem;
+    font-size: 1.72rem;
+    text-align: center;
+    color: var(--mildish-grey);
+    font-weight: 540;
+    margin-bottom: 0;
```
Comment (harsh-rk) [inline signUpPageStyle.css]: Please check if we need this

File: signUpPageStyle.css [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+    border-top: 0.1rem solid var(--mild-grey);
+    width: 10.63rem;
+    margin-left: 1.9rem;
+    margin-top: 1.98rem;
+}
+
+.signup-seperator-text{
+    font-weight: bold;
+    font-size: 1.3rem;
+    color: var(--mildish-grey);
```
Comment (harsh-rk) [inline signUpPageStyle.css]: Please check if we can use the same colour as separator text in login page

File: signUpPageStyle.css [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+
+.login{
+    background: white;
+    border: 0.1rem solid var(--mild-grey);
+    padding: 2rem 3rem;
+    width: 29rem;
+}
+
+.login-text{
+    margin: 0.48rem 0;
```
Comment (harsh-rk) [inline signUpPageStyle.css]: Please check if we need this

File: signUpPage.html [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    </section>
+
+<!--Login page link-->
+    <section class="login">
+        <center><p class="login-text">Have an account? <span class="login-button">Log in</span></p></center>
```
Comment (harsh-rk) [inline signUpPage.html]: We can use an anchor instead of span

File: style.css [hunk 16]
```diff
@@ -253,26 +257,173 @@ section {
...
+.login{
+    background: white;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    border: 0.1rem solid var(--mild-grey);
+    padding: 2rem 3rem;
+    width: 33rem;
+    margin-bottom: 1rem;
```
Comment (harsh-rk) [inline style.css]: Please check if we can remove this

File: style.css [hunk 17]
```diff
@@ -253,26 +257,173 @@ section {
...
+    border-radius: 1rem;
+}
+
+.login-header{
+    display: flex;
+    align-items: flex-start;
+    margin-left: auto;
```
Comment (harsh-rk) [inline style.css]: Please check if we can remove this

File: style.css [hunk 18]
```diff
@@ -253,26 +257,173 @@ section {
...
+
+.login-forgot-password{
+    font-size: 1.25rem;
+    color: var(--facebook-blue);
+    margin-top: 1.5rem;
+    margin-bottom: 3rem;
+}
+
+.login-signup{
+    background: white;
```
Comment (harsh-rk) [inline style.css]: We already have a white background for the entire popup

File: index.html [hunk 19]
```diff
@@ -134,7 +136,49 @@ <h4 class="profile-tabbar-tab-text">TAGGED</h4>
...
+                <p class="login-seperator-text">OR</p>
+                <p class="login-seperator-line-second"></p>
+            </div>
 
+            <div class="login-facebook">
+                <img class="login-facebook-icon" src="./Images/facebook_icon.png">
+                <p class="login-facebook-text">Log in with Facebook</p>
```
Comment (harsh-rk) [inline index.html]: Use anchor tags for
1. Log in with Facebook
2. Forgot password?
3. Sign up

File: index.html [hunk 20]
```diff
@@ -61,14 +62,15 @@
             <img class="profile-tabbar-tab-icon" src="./images/grid.png">
             <h4 class="profile-tabbar-tab-text profile-tabbar-tab-text-selected">POSTS</h4>
         </div>
-        <div class = "profile-tabbar-tab">
+        <div class = "profile-tabbar-tab" data-modal-target="#login-popup-modal">
```
Comment (harsh-rk) [inline index.html]: data-modal-target <> data-login-popup-open-target
And also its counterpart data-login-popup-close-target

File: index.js [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+        currentText.style.backgroundColor = "grey";
+    }
+    else {
+        currentText.innerHTML = "Follow"
+        currentText.style.backgroundColor = "#0595F6";  // Instagram Blue Color
+    }
+})
+
+openModalButtons.forEach(button => {
+    button.addEventListener('click', () => {
```
Comment (harsh-rk) [inline index.js]: This is an inline function, we should avoid it

File: index.js [hunk 22]
```diff
@@ -0,0 +1,40 @@
+const openModalButtons = document.querySelectorAll('[data-modal-target]')
```
Comment (harsh-rk) [inline index.js]: openModalButtons <> loginPopupOpenTargets

File: index.js [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+const closeModalButtons = document.querySelectorAll('[login-popup-close-modal]')
```
Comment (harsh-rk) [inline index.js]: Let's make `js` or `src` folder and create three files:
1. openLoginPopup.js -> default export handleOpenLoginPopup
2. closeLoginPopup.js -> default export handleCloseLoginPopup
3. followButton.js -> default export handleFollowButtonClick

Import all 3 in index.js and call them

File: loginPageStyle.css (hunk 5 above)
Comment (chandak876) [reply loginPageStyle.css]: The `min-height` value assigned to the body element gives it an initial height that matches the HTML element.

File: signUpPageStyle.css (hunk 12 above)
Comment (chandak876) [reply signUpPageStyle.css]: Since we have defined `*root{margin:0}`, browser is taking some default margin.

File: signUpPageStyle.css (hunk 13 above)
Comment (chandak876) [reply signUpPageStyle.css]: Login page is also using the same colour `--mildish-grey`

File: loginPageStyle.css (hunk 5 above)
Comment (chandak876) [reply loginPageStyle.css]: By default both html and body have their height set to `auto`. So setting `height:100%` takes it from viewport and is applied to body as minimum to take up the entire space.

Reference Link - https://stackoverflow.com/questions/17555682/height-100-or-min-height-100-for-html-and-body-elements

File: loginPageStyle.css (hunk 6 above)
Comment (chandak876) [reply loginPageStyle.css]: Height defines the vertical measurement of the component whereas the width property sets an element's horizontal measurement -- according to MDN docs. No reference provided for length in MDN.

**Review by chandak876 — COMMENTED** [review]
Comments Addressed.

File: login-page/loginPage.css [hunk 24]
```diff
@@ -1,14 +1,18 @@
 /*General Styling*/
 :root{
 
+    /*Int Variables*/
+    --container-width: 29rem;
```
Comment (harsh-rk) [inline login-page/loginPage.css]: container-width <> login-container-width

File: login-page/loginPage.css [hunk 25]
```diff
@@ -120,19 +113,20 @@ a{
...
 
 .signup{
     background: white;
     border: 0.1rem solid var(--mild-grey);
     padding: 2rem 3rem;
-    width: 29rem;
+    width: var(--container-width);
 }
 
 .signup-text{
```
Comment (harsh-rk) [inline login-page/loginPage.css:132]: Please remove margin from this class as it is not required

File: src/followButton.js [hunk 26]
```diff
@@ -0,0 +1,16 @@
+const followButton = document.getElementById("profile-content-follow-button")
+
+const followButtonHelper = () => {
+    var currentText = document.getElementById("profile-content-follow-button");
+
+    if(currentText.innerHTML == "Follow") {
+        currentText.innerHTML = "Unfollow";
```
Comment (harsh-rk) [inline src/followButton.js]: Single Responsibility Principle
This function is doing three things right now:
1. Retrieving and comparing the current text of the follow button
2. Does the handling for unfollowing
3. Does the handling for following

Ideally it should only do the first one

File: src/followButton.js [hunk 27]
```diff
@@ -0,0 +1,16 @@
+const followButton = document.getElementById("profile-content-follow-button")
+
+const followButtonHelper = () => {
```
Comment (harsh-rk) [inline src/followButton.js]: followButtonHelper  <> handleFollowButtonClick
handleFollow(followButton), handleUnfollow(followButton)

File: src/closeLoginPopup.js [hunk 28]
```diff
@@ -0,0 +1,30 @@
+const closeLoginPopupOpenTargets = document.querySelectorAll('[data-login-popup-close-target]');
+let loginOpen = 0;
+
+const attachBackgroundListener = (event) => {
```
Comment (harsh-rk) [inline src/closeLoginPopup.js]: Define handleLoginPopupBackgroundClick.js for this and attach it to DOM when popup is opened and remove it when the popup is closed
Separately make a helpers directory for closeLoginPopup.js and openLoginPopup.js
And an event handlers directory for handleOpenLoginPopup.js and handleCloseLoginPopup.js

File: src/closeLoginPopup.js [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+    if(loginOpen == 1 && !event.target.closest('.login')){
+        const modal = document.getElementById('login-popup');
+        closeModal(modal);
+    }
+
+    if(event.target.closest('.profile-tabbar-tab')) loginOpen = 1;
+}
+
+const closeLoginPopupListener = (event) => {
+    const modal = document.querySelector(event.currentTarget.dataset.loginPopupCloseTarget);
```
Comment (harsh-rk) [inline src/closeLoginPopup.js]: modal <> loginPopup

File: src/closeLoginPopup.js [hunk 30]
```diff
@@ -0,0 +1,30 @@
...
+const closeModal = (modal) => {
+    if(modal==null) return;
+
+    modal.classList.remove('active');
+    loginOpen = 0;
+}
+
+window.addEventListener('click', attachBackgroundListener)
+
+closeLoginPopupOpenTargets.forEach(button => {
```
Comment (harsh-rk) [inline src/closeLoginPopup.js]: button <> closeLoginPopupOpenTarget

File: login-page/loginPage.css (hunk 25 above)
Comment (chandak876) [reply login-page/loginPage.css:132]: taking default margins by default which is larger than 0.48 rem.

**Pull request merged**